### PR TITLE
Provide module docstring giving a brief API overview

### PR DESCRIPTION
**Pull request by timholy**
I use this package infrequently and always have to visit readthedocs. This PR is intended to help me (personally :smile:) to get started with just `?Distributions`.

### CODE REVIEW

File: src/Distributions.jl [hunk 1]
```diff
@@ -289,4 +289,49 @@ include("mixtures/unigmm.jl")
...
 include("deprecates.jl")
 
+"""
+A Julia package for probability distributions and associated functions.
+
+API overview (major features):
+
+- `d = Dist(parameters...)` creates a distribution instance `d` for some distribution `Dist` (see choices below) with the specified `parameters`
+- `rand(d, sz)` samples from the distribution
+- `pdf(d, x)` and `logpdf(d, x)` compute the probability or log-probability of `d` at `x`
```
Comment (andreasnoack) [inline src/Distributions.jl]: I'd add "density" here.

**Comment by andreasnoack** [comment]
Thanks for adding this.

**Comment by timholy** [comment]
Done. Note I can't merge this myself, so someone else will have to hit the button.

**Comment by andreasnoack** [comment]
Some changes to type inference have probably caused the test errors on master. These changes are completely unrelated so I'll merge.

**Pull request merged**